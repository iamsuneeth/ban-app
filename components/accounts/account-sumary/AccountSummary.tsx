import React from "react";
import { View, Text, StyleSheet, Dimensions } from "react-native";
import { Card } from "../../elements/card/Card";

export const AccountSummary = () => {
  return (
    <Card style={styles.summary}>
      <View>
        <Text
          style={{
            fontSize: 14,
            color: "gray",
            textTransform: "capitalize"
          }}
        >
          total balance
        </Text>
        <Text
          style={{
            fontSize: 30,
            color: "#333",
            fontWeight: "bold"
          }}
        >
          £3100.00
        </Text>
      </View>
      <View
        style={{
          height: 40
        }}
      >
        <View
          style={{
            flex: 1,
            flexDirection: "row",
            justifyContent: "space-between"
          }}
        >
          <Text
            style={{
              color: "gray",
              fontSize: 14,
              textTransform: "capitalize"
            }}
          >
            available balance
          </Text>

          <Text style={{ color: "#333", fontSize: 14 }}>£5100.00</Text>
        </View>
        <View
          style={{
            flex: 1,
            flexDirection: "row",
            justifyContent: "space-between"
          }}
        >
          <Text
            style={{
              color: "gray",
              fontSize: 14,
              textTransform: "capitalize"
            }}
          >
            overdfart
          </Text>

          <Text
            style={{
              color: "#333",
              fontSize: 14
            }}
          >
            £2000.00
          </Text>
        </View>
      </View>
    </Card>
  );
};

const styles = StyleSheet.create({
  summary: {
    flex: 1,
    justifyContent: "space-evenly",
    width: Dimensions.get("window").width - 20,
    margin: 10,
    padding: 10
  }
});
