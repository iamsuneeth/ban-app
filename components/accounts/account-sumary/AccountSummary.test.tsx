import React from "react";
import { render } from "@testing-library/react-native";
import { AccountSummary } from "./AccountSummary";

describe("tests for AccountSummary component", () => {
  test("should render correctly", () => {
    const { baseElement } = render(<AccountSummary />);
    expect(baseElement).toMatchSnapshot();
  });
});
