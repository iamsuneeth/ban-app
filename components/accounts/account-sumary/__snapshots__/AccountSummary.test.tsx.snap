// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for AccountSummary component should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 5,
          "flex": 1,
          "justifyContent": "space-evenly",
          "margin": 10,
          "padding": 10,
          "shadowOffset": Object {
            "height": 2,
            "width": 5,
          },
          "width": 730,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "gray",
              "fontSize": 14,
              "textTransform": "capitalize",
            }
          }
        >
          total balance
        </Text>
        <Text
          style={
            Object {
              "color": "#333",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          £3100.00
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 40,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "gray",
                "fontSize": 14,
                "textTransform": "capitalize",
              }
            }
          >
            available balance
          </Text>
          <Text
            style={
              Object {
                "color": "#333",
                "fontSize": 14,
              }
            }
          >
            £5100.00
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "gray",
                "fontSize": 14,
                "textTransform": "capitalize",
              }
            }
          >
            overdfart
          </Text>
          <Text
            style={
              Object {
                "color": "#333",
                "fontSize": 14,
              }
            }
          >
            £2000.00
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
